src/gba/apu.nim

```diff
@@ -75,14 +75,8 @@ proc getSample(apu: APU): proc() = (proc() =
 proc tickFrameSequencer(apu: APU): proc() = (proc() =
   apu.gba.scheduler.schedule(frameSequencerPeriod, apu.tickFrameSequencer(), EventType.apu)
   case frameSequencerStage
-  of 0:
-    cast[Channel1](apu.channel1).lengthStep()
-  of 2:
-    cast[Channel1](apu.channel1).lengthStep()
-  of 4:
-    cast[Channel1](apu.channel1).lengthStep()
-  of 6:
-    cast[Channel1](apu.channel1).lengthStep()
+  of 0, 2, 4, 6:
+    Channel1(apu.channel1).lengthStep()
```

Comment:
Once the other channels are added, this will just need to be split again

Reply:
Yea this was something i was questioning about, didnt know if this was lacking implementation or just not knowing about switch cases having multiple options